MINOR: Update jacoco to 0.8.7 for JDK 16 support

ijuma opened the pull request:
Details:
* https://github.com/jacoco/jacoco/releases/tag/v0.8.6
* https://github.com/jacoco/jacoco/releases/tag/v0.8.7

Ran `./gradlew clients:reportCoverage -PenableTestCoverage=true -Dorg.gradle.parallel=false`
successfully with Java 15 (see https://github.com/gradle/gradle/issues/15730 and
https://github.com/scoverage/gradle-scoverage/issues/150 for the reason why 
`-Dorg.gradle.parallel=false` is required).

Also updated `README.md` to include `-Dorg.gradle.parallel=false` alongside `reportCoverage`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

ijuma committed:
MINOR: Update jacoco to 0.8.7 for JDK 16 support

Details: https://github.com/jacoco/jacoco/releases/tag/v0.8.7

ijuma committed:
Use 0.8.7 for real

chia7712 reviewed (approved):
LGTM

> Ran ./gradlew clients:reportCoverage -PenableTestCoverage=true -Dorg.gradle.parallel=false
successfully with Java 15 (see gradle/gradle#15730 and
scoverage/gradle-scoverage#150 for the reason why
-Dorg.gradle.parallel=false is required).

Could you add this to `README`? The instruction in `README` is `./gradlew reportCoverage -PenableTestCoverage=true` and it does not work.

ijuma committed:
Add `-Dorg.gradle.parallel=false` workaround for `reportCoverage` in README

https://github.com/gradle/gradle/issues/15730 and
https://github.com/scoverage/gradle-scoverage/issues/150 have details on
why `-Dorg.gradle.parallel=false` is required.
